USCSandbox/Program.cs

```diff
@@ -34,16 +54,39 @@ static void Main(string[] args)
                 manager.LoadClassPackage("classdata.tpk");
                 manager.LoadClassDatabaseFromPackage(afile.Metadata.UnityVersion);
                 ver = UnityVersion.Parse(afile.Metadata.UnityVersion);
+                shaderTypeId = manager.ClassPackage.GetClassDatabase(ver.ToString()).FindAssetClassByName("Shader").ClassId;
             }
 
-            var shaderBf = manager.GetExtAsset(afileInst, 0, shaderPathId).baseField;
-            if (shaderBf == null)
+            var shaders = afileInst.file.GetAssetsOfType(shaderTypeId);
+            int unnamedCount = 0;
+            foreach (var shader in shaders)
             {
-                Console.WriteLine("Shader asset not found.");
-                return;
+                if (shaderPathId != -1 && shader.PathId != shaderPathId)
+                    continue;
+                
+                var shaderBf = manager.GetExtAsset(afileInst, 0, shader.PathId).baseField;
+                if (shaderBf == null)
+                {
+                    Console.WriteLine("Shader asset not found.");
+                    return;
+                }
+                var shaderProcessor = new ShaderProcessor(shaderBf, ver, shaderPlatform);
+                bool fileNameExists = files.TryGetValue(shader.PathId, out string? name);
```

Comment:
You don't need the container string for the shader name. Just used m_ParsedForm:
```suggestion
                var name = shaderBf["m_ParsedForm"]["m_Name"].AsString;
```

Reply:
This is incorrect: I want I to replicate what assetripper did here because it allows me to very quickly copy paste the file into the unity project and test them. This is also not practical to do because these are how the shaders are named in editor, but NOT how it is placed in the project's folder which is what I care about here.

Reply:
I would at least suggest it as a fallback. Most of the shaders I tested didn't have a container causing me to have hundreds of "unnamed shader" files

Reply:
The unanmed are actually unity's integraded shaders

They don't matter because a project never has them, it just comes with the editor and is compiled in the assets

Reply:
Ended up adding a falback for those